src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/ReportService.kt

```diff
@@ -187,62 +188,64 @@ class ReportService(
   }
 
   @Transactional
-  fun changeReportStatus(id: UUID, changeStatusRequest: ChangeStatusRequest): Pair<ReportWithDetails, Boolean>? {
-    return reportRepository.findById(id).map {
-      val changed = it.status != changeStatusRequest.newStatus
-      if (changed) {
+  fun changeReportStatus(id: UUID, changeStatusRequest: ChangeStatusRequest): MaybeChanged<ReportWithDetails>? {
+    return reportRepository.findById(id).getOrNull()?.let { reportEntity ->
+      val maybeChangedReport = if (reportEntity.status != changeStatusRequest.newStatus) {
         // TODO: determine which transitions are allowed
 
         val now = LocalDateTime.now(clock)
         val user = authenticationFacade.getUserOrSystemInContext()
-        it.changeStatus(
+        reportEntity.changeStatus(
           newStatus = changeStatusRequest.newStatus,
           changedAt = now,
           changedBy = user,
         )
-        it.modifiedAt = now
-        it.modifiedBy = user
+        reportEntity.modifiedAt = now
+        reportEntity.modifiedBy = user
+
+        MaybeChanged.Changed(reportEntity.toDtoWithDetails())
+      } else {
+        MaybeChanged.Unchanged(reportEntity.toDtoWithDetails())
       }
 
-      it.toDtoWithDetails().apply {
-        if (changed) {
-          log.info("Changed incident report status to ${changeStatusRequest.newStatus} for number=$incidentNumber ID=$id")
-          telemetryClient.trackEvent(
-            "Changed incident report status",
-            this,
-          )
-        }
-      } to changed
-    }.getOrNull()
+      maybeChangedReport.alsoIfChanged { reportWithDetails ->
+        log.info("Changed incident report status to ${changeStatusRequest.newStatus} for number=${reportWithDetails.incidentNumber} ID=$id")
```

Comment:
I still think it could be moved in the if block above :)

Reply:
the event publishing thing isn't available in the service and i'm not sure i can wire it up properly :(

it's weird that the other service methods already had logging and telemetry in the service itself but event publishing outside it in the resource